src/compiler/utilities.ts

```diff
@@ -10639,3 +10639,23 @@ export function replaceFirstStar(s: string, replacement: string): string {
 export function getNameFromImportAttribute(node: ImportAttribute) {
     return isIdentifier(node.name) ? node.name.escapedText : escapeLeadingUnderscores(node.name.text);
 }
+
+/** @internal */
+export function isSyntacticallyString(expr: Expression): boolean {
+    switch (expr.kind) {
+        case SyntaxKind.BinaryExpression:
+            const left = (expr as BinaryExpression).left;
+            const right = (expr as BinaryExpression).right;
+            return (
+                (expr as BinaryExpression).operatorToken.kind === SyntaxKind.PlusToken &&
+                (isSyntacticallyString(left) || isSyntacticallyString(right))
+            );
+        case SyntaxKind.ParenthesizedExpression:

```

Comment:
Technically I think this might need the skip parents function for this as I think these also need to handle type casts in JS via JSDoc?